ePermit-KLF/web_flask/app.py

```diff
@@ -26,28 +28,48 @@ def index():
 @app.route('/signup', methods=['GET', 'POST'], strict_slashes=False)
 def sign_up():
     if request.method == 'POST':
-        user = User(email=request.form.get("email"),
-                    password=request.form.get("password"))
-        if request.form.get('password') != request.form.get("confirm_password"):
+        email = request.form.get("email")
+        password = request.form.get("password")
+        confirm_password = request.form.get('confirmed_password')
+
+        # Check if email exixts in db
+        users = storage.all(User)
+        print(users)
+        for user in users:
```

Comment:
emails can never be same, hii error check has no importance ju emails are ever uniq, the check should be if the email is a valid email
usernames ndio we do this checks

Reply:
Which  error check, the one stating if user.email == email was to check if the email the user is trying to use already exist in the database, so instead of them signing up  again they are redirected to signing in